src/components/jobs/create/jobCreateScheduleDialog.ts

```diff
@@ -64,11 +65,50 @@ export class JobScheduleDialog {
             this.formState.schedule = true;
         }
 
-        if (!this.job.offset && this.job.offset !== 0) {
+        if ((!this.job.offset && this.job.offset !== 0) || this.job.offset < 0) {
             error = true;
             this.formState.offset = false;
         } else {
-            this.formState.offset = true;
+            if (this.job.schedule_id) {
+                let job_schedule = null;
+                this.schedules.forEach(schedule => {
```

Comment:
This loop will execute for all schedules even after it finds one. Not a huge deal given the sample size, but we should at least break, or switch to something like a lodash find function.